rocAL/rocAL_pybind/examples/rocAL_api_coco_pipeline.py

```diff
@@ -58,170 +57,169 @@ def __init__(self, pipelines, tensor_layout=types.NCHW, reverse_channels=False,
+                self.dimensions = self.output_tensor_list[i].dimensions()
+                self.torch_dtype = self.output_tensor_list[i].dtype()
+                if self.device == "cpu":
+                    self.output = torch.empty(self.dimensions, dtype=getattr(torch, self.torch_dtype))
+                else:
+                    torch_gpu_device = torch.device('cuda', self.device_id)
+                    self.output = torch.empty(self.dimensions, dtype=getattr(torch, self.torch_dtype), device=torch_gpu_device)
+                self.output_tensor_list[i].copy_data(ctypes.c_void_p(self.output.data_ptr()), self.output_memory_type)
+                self.output_list.append(self.output)
+        else:
+            for i in range(len(self.output_tensor_list)):
+                self.output_tensor_list[i].copy_data(ctypes.c_void_p(self.output_list[i].data_ptr()), self.output_memory_type)
+
+        self.labels = self.loader.get_bounding_box_labels()
+        # 1D bboxes array in a batch
+        self.bboxes = self.loader.get_bounding_box_cords()
+        self.loader.get_image_id(self.image_id)
+        image_id_tensor = torch.tensor(self.image_id)
+        image_size_tensor = torch.tensor(self.img_size).view(-1, self.bs, 2)
+
+        for i in range(self.bs):
+            if self.display:
+                img = self.output
+                draw_patches(img[i], self.image_id[i],
+                             self.bboxes[i], self.device)
+        return (self.output), self.bboxes, self.labels, image_id_tensor, image_size_tensor
 
     def reset(self):
-        self.loader.rocalResetLoaders()
+        self.loader.rocal_reset_loaders()
 
     def __iter__(self):
         return self
 
-def draw_patches(img, idx, bboxes, device="cpu"):
+def draw_patches(img, idx, bboxes, device):
+    args = parse_args()
+    # image is expected as a tensor, bboxes as numpy
     import cv2
     if device == "cpu":
         image = img.detach().numpy()
     else:
         image = img.cpu().numpy()
-    image = image.transpose([1, 2, 0])
+    if args.fp16:
+        image = (image).astype('uint8')
+    if not args.NHWC:
+        image = image.transpose([1, 2, 0])
+    image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+    bboxes = np.reshape(bboxes, (-1, 4))
 
     for (l, t, r, b) in bboxes:
         loc_ = [l, t, r, b]
         color = (255, 0, 0)
         thickness = 2
-        image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
         image = cv2.UMat(image).get()
         image = cv2.rectangle(image, (int(loc_[0]), int(loc_[1])), (int(
             (loc_[2])), int((loc_[3]))), color, thickness)
-        cv2.imwrite("OUTPUT_IMAGES_PYTHON/COCO_READER/" + str(idx)+"_"+"train"+".png", image * 255)
+        cv2.imwrite("OUTPUT_IMAGES_PYTHON/COCO_READER/" +
+                    str(idx)+"_"+"train"+".png", image)
+
 
 def main():
-    if  len(sys.argv) < 3:
-        print ('Please pass image_path annotation_path cpu/gpu batch_size')
-        exit(0)
+    args = parse_args()
```

Comment:
@SundarRajan28 we still want to tell the user what to pass with a print if args are not correct.